processing/src/main/java/org/apache/druid/segment/virtual/NestedFieldVirtualColumn.java

```diff
@@ -647,7 +650,10 @@ public ColumnCapabilities capabilities(ColumnInspector inspector, String columnN
     final ColumnCapabilities capabilities = inspector.getColumnCapabilities(this.columnName);
 
     if (capabilities != null) {
-      if (!capabilities.isPrimitive() && capabilities.isDictionaryEncoded().isTrue()) {
+      // if the underlying column is a nested column (and persisted to disk, re: the dictionary encoded check)
```

Comment:
Do we really have no better way of determining if we have been persisted than checking `isDictionaryEncoded()`?  Given that we are already a `NestedFieldVirtualColumn` and supposedly working with a `NestedField` that we understand the class of, I wonder if we couldn't have something more specific.  Maybe for another day, but perhaps we could have a `NestedFieldColumnCapabilities` that we could cast to and get access to more specific methods?

Reply:
yeah, im certain there is a cooler way to do this, especially when the underlying inspector is the segment, but i'll save that for another day